examples/colors/main.rs

```diff
@@ -0,0 +1,169 @@
+        model.set_property("size", 0u32);
+        _btn.add_tick_callback(clone!(@strong model => move |_, frame| {
+            let limit = model.limit();
+
+            model.set_property("size", limit.min(model.size() + 1.max(limit / 4096)));
+
+            return Continue(if (model.size() >= limit) {
+              false
+            } else {
+              true
+            });
+        }));
+    }));
+    header.pack_start(&refill_btn);
+
+    let progress = gtk::ProgressBar::new();
+    progress.set_hexpand(true);
+    progress.set_valign(gtk::Align::Start);
+    progress.set_visible(false);
+
+    sort_model.connect_pending_notify(clone!(@strong progress => move |model| {
+        let total = model.n_items();
+        let total = total.max(1); /* avoid div by 0 below */
+        let pending = model.pending();
+
+        progress.set_visible(pending != 0);
+        progress.set_fraction((total - pending) as f64 / total as f64);
+    }));
+
+    let label = gtk::Label::new(Some("0 /"));
+    let attrs = gtk::pango::AttrList::new();
+    attrs.insert(gtk::pango::AttrFontFeatures::new("tnum").upcast());
+    label.set_attributes(Some(&attrs));
+    label.set_xalign(1.0);
+    header.pack_start(&label);
+
+    let dropdown = gtk::DropDown::from_strings(&list);
+    dropdown.connect_selected_notify(clone!(@strong selected, @strong label => move |dropdown| {
+        let num_str = list[dropdown.selected() as usize];
+        if let Ok(num) = num_str.parse::<u32>() {
+            model.set_property("size", num);
+            model.set_property("limit", num);
+
+            label.set_label(&format!("{} /", num_str));
+
+            *selected.borrow_mut() = num;
+        }
+    }));
+    dropdown.set_selected(3);
+    header.pack_start(&dropdown);
+
+    let list = [
+        "Unsorted",
+        "Hue",
+        "Saturation",
+        "Value",
+        "Red",
+        "Green",
+        "Blue",
+    ];
+    let dropdown = gtk::DropDown::from_strings(&list);
+    dropdown.connect_selected_notify(clone!(@strong sort_model, @strong list => move |a| {
+        let sorter = match &list[a.selected() as usize].to_lowercase() {
+            name => {
+                let sorter = gtk::NumericSorter::new(Some(&gtk::PropertyExpression::new(Color::static_type(), None::<&gtk::Expression>, name)));
+                sorter.set_sort_order(gtk::SortType::Descending);
+                Some(sorter)
+            },
+            _ => None
+        };
+        sort_model.set_sorter(sorter.as_ref());
+    }));
+
+    header.pack_end(&dropdown);
+    header.pack_end(&gtk::Label::new(Some("Sort By: ")));
+
+    let grid_view = gtk::GridView::new(Some(&selection), Some(&factory));
+    grid_view.set_hscroll_policy(gtk::ScrollablePolicy::Natural);
+    grid_view.set_vscroll_policy(gtk::ScrollablePolicy::Natural);
+
+    grid_view.set_max_columns(24);
+    grid_view.set_enable_rubberband(true);
```

Comment:
```suggestion
    let grid_view = gtk::GridView::new(Some(&selection), Some(&factory));
    grid_view.set_hscroll_policy(gtk::ScrollablePolicy::Natural);
    grid_view.set_vscroll_policy(gtk::ScrollablePolicy::Natural);

    grid_view.set_max_columns(24);
    grid_view.set_enable_rubberband(true);
```
perfect use case for the builder pattern